### PR TITLE
fix: add cache for getting avatar meta from kv at avatar plugin

### PR DESCRIPTION
**Pull request by lelenei**
Closes mf-1383

### CODE REVIEW

File: packages/mask/src/plugins/Avatar/hooks/usePersonaNFTAvatar.ts [hunk 1]
```diff
@@ -1,25 +1,47 @@
...
 import { NetworkPluginID } from '@masknet/web3-shared-base'
+import addSeconds from 'date-fns/addSeconds'
 import { useAsyncRetry } from 'react-use'
 import { activatedSocialNetworkUI } from '../../../social-network'
 import type { RSS3_KEY_SNS } from '../constants'
+import type { AvatarMetaDB, NextIDAvatarMeta } from '../types'
 import { getNFTAvatarByUserId } from '../utils'
 import { useGetNFTAvatar } from './useGetNFTAvatar'
 
+const cache = new Map<string, [Promise<NextIDAvatarMeta | undefined>, number]>()
```
Comment (UncleBill) [inline packages/mask/src/plugins/Avatar/hooks/usePersonaNFTAvatar.ts]: Recommend using `lru-cache`.

File: packages/mask/src/plugins/Avatar/hooks/usePersonaNFTAvatar.ts [hunk 2]
```diff
@@ -3,23 +3,53 @@ import { NetworkPluginID } from '@masknet/web3-shared-base'
...
-        )
-        if (!avatarMeta) return
-        if (avatarMeta.pluginId === NetworkPluginID.PLUGIN_SOLANA) {
-            return { imageUrl: '', nickname: '', ...avatarMeta, address: avatarMeta.tokenId }
+        if (!userId) return
+        const key = `${userId}-${activatedSocialNetworkUI.networkIdentifier}`
+        let v = cache.get(key)
+        if (!v) {
+            cache.set(key, getNFTAvatarForCache(userId, snsKey, avatarId, getNFTAvatar))
         }
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Avatar/hooks/usePersonaNFTAvatar.ts]:
```suggestion
        const v = cache.get(key) ?? cache.set(key, getNFTAvatarForCache(userId, snsKey, avatarId, getNFTAvatar))
```

Comment (lelenei) [reply packages/mask/src/plugins/Avatar/hooks/usePersonaNFTAvatar.ts]: typescript error

File: packages/mask/src/plugins/Avatar/hooks/usePersonaNFTAvatar.ts [hunk 3]
```diff
@@ -3,23 +3,50 @@ import { NetworkPluginID } from '@masknet/web3-shared-base'
...
-            return { imageUrl: '', nickname: '', ...avatarMeta, address: avatarMeta.tokenId }
-        }
-        return { imageUrl: '', nickname: '', ...avatarMeta }
-    }, [userId, getNFTAvatar, avatarId, snsKey])
+        if (!userId) return
+        const key = `${userId}-${activatedSocialNetworkUI.networkIdentifier}`
+        let v = cache.get(key)
+        if (!v) cache.set(key, getNFTAvatarForCache(userId, snsKey, avatarId, getNFTAvatar))
+        v = cache.get(key)
+        return v
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Avatar/hooks/usePersonaNFTAvatar.ts]:
```suggestion
        if (!cache.has(key)) cache.set(key, getNFTAvatarForCache(userId, snsKey, avatarId, getNFTAvatar))
        return cache.get(key)!
```

File: packages/mask/src/plugins/Avatar/hooks/usePersonaNFTAvatar.ts [hunk 4]
```diff
@@ -3,23 +3,48 @@ import { NetworkPluginID } from '@masknet/web3-shared-base'
...
-        )
-        if (!avatarMeta) return
-        if (avatarMeta.pluginId === NetworkPluginID.PLUGIN_SOLANA) {
-            return { imageUrl: '', nickname: '', ...avatarMeta, address: avatarMeta.tokenId }
-        }
-        return { imageUrl: '', nickname: '', ...avatarMeta }
-    }, [userId, getNFTAvatar, avatarId, snsKey])
+        if (!userId) return
+        const key = `${userId}-${activatedSocialNetworkUI.networkIdentifier}`
+        if (!cache.get(key)) cache.set(key, getNFTAvatarForCache(userId, snsKey, avatarId, getNFTAvatar))
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Avatar/hooks/usePersonaNFTAvatar.ts]:
```suggestion
        if (!cache.has(key)) cache.set(key, getNFTAvatarForCache(userId, snsKey, avatarId, getNFTAvatar))
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**